### PR TITLE
Sync FF2 with SM version

### PR DESCRIPTION
**Pull request by JassJam**
#172.

Since @Pheubel and I did almost the same thing, my approach was to unload all plugins during load/unload to load only 
the needed plugins during an active round, well besides that some syntaxes fix and I've had problems with some subplugins that were using some natives like `FF2_GetRoundState` which call `VSH2GameMode.GetPropInt("iRoundState")` too early, now it's checking if vsh2 is already running before proceeding to do anything else.

Updated `GetThisPluginName`, changed extension name from '.ff2' to .'smx'.

### CODE REVIEW

**Comment by Pheubel** [comment]
i just had a look at your PR, overall it seems to be of a better quality than mine, there is however one concern i have with the current solution for replacing ".ff2" in your PR for edge cases, for example: lets say someone made a plugin named "super_ability.ff2.stable.ff2" (dumb name, yeah ik) and a boss that uses it, what would happen is that it would turn it into "super_ability.smx.stable.smx" after which the config made for that boss can become misalligned. i don't know how well FF2 can handle that, but that is about it

**Comment by JassJam** [comment]
Sounds good to me

File: addons/sourcemod/scripting/modules/ff2/subplugins.sp [hunk 1]
```diff
@@ -100,17 +100,78 @@ methodmap FF2PluginList < ArrayList {
...
+		char plugin_buffer[PLATFORM_MAX_PATH];
+		char renamed_plugin_buffer[PLATFORM_MAX_PATH];
+		while( plugin_directory.GetNext(plugin_buffer, PLATFORM_MAX_PATH, file_type) ) {
+			if( file_type != FileType_File ) {
+				continue;
+			}
+
+			/// make sure the file ends with ".ff2"
+			int extension_index = FindCharInString(plugin_buffer, '.', true);
+			if( extension_index == -1 || extension_index + 3 > PLATFORM_MAX_PATH || plugin_buffer[extension_index+1] != 'f'|| plugin_buffer[extension_index+2] != 'f' || plugin_buffer[extension_index+3] != '2' ) {
```
Comment (Pheubel) [inline addons/sourcemod/scripting/modules/ff2/subplugins.sp]: i think that by replacing `extension_index + 3 > PLATFORM_MAX_PATH` by `extension_index > FF2_MAX_PLUGIN_NAME - 1` it would make sure that the plugin name is at most 63 characters (64th reserved for the null terminator).

however to make it sure that the extension is .ff2 and not .ff2andsomeothercharacters the character at the 4th position after the extension index needs to be tested for a null character like this `plugin_buffer[extension_index+4] != 0`.

so the full line would be:
`if( extension_index == -1 || extension_index > FF2_MAX_PLUGIN_NAME - 1 || plugin_buffer[extension_index+1] != 'f'|| plugin_buffer[extension_index+2] != 'f' || plugin_buffer[extension_index+3] != '2' || plugin_buffer[extension_index+4] != 0 ) {`

**Review by Pheubel — CHANGES_REQUESTED** [review]
In my code solving this issue i made a small mistake, i forgot to keep the terminating character in mind and that the maximum length of a plugin is 64 characters (as seen with FF2_MAX_PLUGIN_NAME ). 

sorry for not catching this earlier.

File: addons/sourcemod/scripting/modules/ff2/subplugins.sp (hunk 1 above)
Comment (JassJam) [reply addons/sourcemod/scripting/modules/ff2/subplugins.sp]: Updated
